crates/starknet-server/tests/test_restart.rs

```diff
@@ -0,0 +1,255 @@
+// must use `pub`: https://github.com/rust-lang/rust/issues/46379#issuecomment-548787629
+pub mod common;
+
+mod test_restart {
+    use std::path::Path;
+    use std::sync::Arc;
+
+    use hyper::StatusCode;
+    use starknet_core::constants::{CAIRO_0_ACCOUNT_CONTRACT_HASH, ERC20_CONTRACT_ADDRESS};
+    use starknet_core::utils::exported_test_utils::dummy_cairo_0_contract_class;
+    use starknet_rs_accounts::{
+        Account, AccountFactory, ExecutionEncoding, OpenZeppelinAccountFactory, SingleOwnerAccount,
+    };
+    use starknet_rs_core::chain_id;
+    use starknet_rs_core::types::contract::legacy::LegacyContractClass;
+    use starknet_rs_core::types::{BlockId, BlockTag, FieldElement, StarknetError};
+    use starknet_rs_core::utils::get_storage_var_address;
+    use starknet_rs_providers::{
+        MaybeUnknownErrorCode, Provider, ProviderError, StarknetErrorWithMessage,
+    };
+
+    use crate::common::constants::CHAIN_ID;
+    use crate::common::devnet::BackgroundDevnet;
+    use crate::common::utils::{get_deployable_account_signer, remove_file, send_ctrl_c_signal};
+
+    #[tokio::test]
+    async fn assert_restartable() {
+        let devnet = BackgroundDevnet::spawn().await.unwrap();
+        let resp = devnet.restart().await.unwrap();
+        assert_eq!(resp.status(), StatusCode::OK);
+    }
+
+    #[tokio::test]
+    async fn assert_tx_not_present_after_restart() {
+        let devnet = BackgroundDevnet::spawn().await.unwrap();
+
+        // generate dummy tx
+        let mint_hash = devnet.mint(FieldElement::ONE, 100).await;
+        assert!(devnet.json_rpc_client.get_transaction_by_hash(mint_hash).await.is_ok());
+
+        let restart_resp = devnet.restart().await.unwrap();
+        assert_eq!(restart_resp.status(), StatusCode::OK);
+
```

Comment:
We could also check if there is no block after restart.

Reply:
Added in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/238/commits/8c566b5e6b2ac654f66e90baedb4096c1383969d